src/vantage6_strongaya_general/privacy_measures.py

```diff
@@ -0,0 +1,1427 @@
+) -> float:
+    """
+    Generate noise based on the chosen differential privacy mechanism.
+
+    Using a consistent seed ensures the same query always produces the same noise,
+    preventing multiple query averaging attacks while maintaining privacy guarantees.
+
+    Args:
+        sensitivity (float): The sensitivity of the query.
+        epsilon (float): The privacy parameter (smaller means more privacy).
+        delta (float): The failure probability parameter (used only for Gaussian).
+        distribution (NoiseDistribution): The DP mechanism to use:
+            - 'laplace': Provides pure ε-differential privacy
+            - 'gaussian': Provides (ε,δ)-differential privacy
+        seed (Optional[int]): Seed for the random number generator, used for deterministic noise.
+            If provided, the same seed will always produce the same noise value.
+
+    Returns:
+        float: A noise value from the appropriate distribution.
+    """
+    # Create a deterministic RNG if seed is provided
+    if seed is not None:
+        # Create a separate random number generator that won't affect other code
+        rng = np.random.RandomState(seed)
+    else:
+        # Use the global RNG if no seed (less secure but maintains backwards compatibility)
+        rng = np.random.RandomState()
+
+    if distribution == "laplace":
+        scale = _get_laplace_scale(sensitivity, epsilon)
+        return rng.laplace(0, scale)
+    else:  # gaussian
+        sigma = _get_gaussian_noise_scale(sensitivity, delta, epsilon)
+        return rng.normal(0, sigma)
+
+
+def _sanitise_metadata(metadata: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Remove potentially sensitive information from metadata.
+
+    Args:
+        metadata (Dict[str, Any]): Original metadata dictionary
+
+    Returns:
+        Dict[str, Any]: Sanitised metadata dictionary
+    """
+    # Create a copy to avoid modifying the original
+    safe_metadata = metadata.copy()
+
+    # Remove or obfuscate sensitive fields
+    if "sensitivity" in safe_metadata:
+        # Instead of exact sensitivity, use a range indicator
+        sensitivity = safe_metadata["sensitivity"]
+        if sensitivity < 0.01:
+            safe_metadata["sensitivity_range"] = "very low"
+        elif sensitivity < 0.1:
+            safe_metadata["sensitivity_range"] = "low"
+        elif sensitivity < 1.0:
+            safe_metadata["sensitivity_range"] = "medium"
+        else:
+            safe_metadata["sensitivity_range"] = "high"
+
+        del safe_metadata["sensitivity"]
+
+    # Other potentially sensitive fields to remove
+    sensitive_fields = ["bin_edges", "categories"]
+    for field in sensitive_fields:
+        if field in safe_metadata:
+            # Instead of values, just store the count
+            if isinstance(safe_metadata[field], list):
+                safe_metadata[f"{field}_count"] = len(safe_metadata[field])
+            del safe_metadata[field]
+
+    return safe_metadata
```

Comment:
To make this more easily maintainable (and more easily readable), consider creating dataclasses and/or enums to make it explicit what the metadata structure is and which variables are deemed sensitive (and thus modified by this function)